src/jazkarta/easyformplugin/savecontent/browser.py

```diff
@@ -16,3 +26,35 @@ def has_saved_content(self):
 
     def saved_content_url(self):
         return self.context.absolute_url() + '/' + STORAGE_ID
+
+    def csv_download_url(self):
+        return self.context.absolute_url() + '/' + STORAGE_ID + '/' + 'download-csv'
+
+
+class CSVDownload(BrowserView):
+
+    def __call__(self):
+        form = self.context.aq_parent
+        items_dicts = []
+        schema_fields = OrderedDict()
+        schema = get_schema(form)
+        for field_name in schema:
+            schema_fields[field_name] = schema.get(field_name).title
+            if isinstance(schema_fields[field_name], unicode):
+                schema_fields[field_name] = schema_fields[field_name].encode("utf-8")
+        for obj in self.context.objectValues():
+            obj_dict = {}
+            for field_name, field_title in schema_fields.items():
+                obj_dict[field_title] = getattr(obj, field_name, '')
+                if isinstance(obj_dict[field_title], unicode):
```

Comment:
I think we need to do something here to handle non-unicode values. Maybe `json.dumps(IJsonCompatible(obj_dict[field_title]))`?

Reply:
I think maybe we should be using `safe_nativestring` here for encoding strings (now importable from `mtneers.content.utils`) so we don't have to redo this for Python 3.

Reply:
I don't think we should import `mtneers.content` code from `jazkarta.easyformplugin.savecontent`.

Reply:
@silviot indeed! Because of that I also added the function into master in `jazkarta.easyformplugin.savecontent.browser`